### DISCUSSION TITLE
Finally I did it!

### DISCUSSION THREAD

**Discussion by derjoerg** [kingsleyadam/local-abbfreeathome-hass#118] [Show and tell]
I just wanted to tell an important milestone (for me :wink:)

After several night shifts I finally migrated my production HA to this integration and deactivated the famous freeathome integration.
My setup of F@H includes 84 devices with 264 entities.

I needed to modify around 20 dashboards, 15 automations, reams of helpers and some scripts.

After a week with the new setup I see no roadblocks and the latency is good.

❤️

**Comment by Alox-K** [comment]
Sounds Great, congrats! Im also eager to do this step, but i need to wait for some missing devices…

**Comment by kingsleyadam** [comment]
Nice! Congrats! Sounds like a lot of work, glad it went well.

**Reply by derjoerg** [reply]
@kingsleyadam: Absolutely!

Nice, that you are back.

I was a little afraid that we lost you 😄

**Reply by kingsleyadam** [reply]
Ha, I was completely offline/checked out for the holidays as I travel for family. And I got sick once I got home so I couldn't even look at a screen. No worries, I'm still around. :)

**Comment by derjoerg** [comment]
Just as a sidenote:
The current integration is called `abbfreeathome_ci`. Whenever this is integrated into the core (🤞 ), it will be renamed to `abbfreeathome`.

What does that mean:
Any modifications (renaming of entitiy_ids, names and so on) you are doing now with the custom integration needs to be redone as soon as the core integration is released.

There is a way around this, !!! BUT !!! this is absolutely NOT supported and NOT recommended by the HA-developers!!!

So, you have been warned 😉 

1. You have to follow this repository to get informed about new releases as you can't use HACS anymore
2. Identify a location where you can execute `git` (if you don't know how to work with `git`, you should stop here)
2.1. e.g. I have created a download-directory directly in the config-directory of my HA-installation
3. Clone the current main branch of this repository into this directory
4. Copy the folder `abbfreeathome_ci` into the `custom_components` folder of your HA-installation
5. Rename the folder from `abbfreeathome_ci` to `abbfreeathome`
6. Modify two files in this folder
6.1. `manifest.json`: Change the value of the key `domain` from `abbfreeathome_ci` to `abbfreeathome`
6.2. `const.py`: Change the value of the constant `domain` from `abbfreeathome_ci` to `abbfreeathome`
7. Restart HA

Drawback:
- There will be no integration-icons (ABB) shown

On every update of the integration:
1. Do a `git pull` within the initial directory (see 2. above)
2. Replace all files within `custom_components/abbfreeathome` with the updated version
3. Redo the modifications (see 6. above)

As soon as the core-integration is released nothing should (!!! as stated, no support !!!) change and the custom integration still has priority over the core-integration.
At the beginning, not all devices will be directly supported from the core-integration. Support will be added step-by-step.
As soon as you think you can switch to the core-integration, simply delete the folder within `custom_components` and restart HA.
